engine/src/main/java/org/terasology/rendering/opengl/GLSLShader.java

```diff
@@ -279,25 +274,36 @@ private void updateAvailableFeatures() {
         }
     }
 
-    private void compileShaders(Set<ShaderProgramFeature> features) {
-        compileShader(GL20.GL_FRAGMENT_SHADER, features);
-        compileShader(GL20.GL_VERTEX_SHADER, features);
-    }
-
     private void compileAllShaderPermutations() {
```

Comment:
I looked into the code a bit better and I can see it holds two maps, one for vertex the other for fragment shaders, using a featureHash as key and the shaderId returned by glCreateShader() as value. So, the guess in the previous comment of mine is incorrect. Still, it'd be good to explain here or even in the javadoc for public method recompile() what this is doing.

But perhaps this is all out of the scope of this PR. I feel for example that an instance of this class is not, in fact, a shader in the opengl/computer graphic sense of the word, but the manager for a number of variants of the same shader. I also wonder if it's worth to keep all those variants in GPU memory compared to recompiling a shader on the fly when a feature is activated/deactivated. But as I said, this is probably out of the scope of this particular PR.